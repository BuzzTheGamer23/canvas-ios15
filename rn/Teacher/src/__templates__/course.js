//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'
import { section } from './section'

export const course: Template<Course> = template({
  id: '1',
  name: 'Learn React Native',
  short_name: 'rn',
  course_code: 'rn 101',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  is_favorite: true,
  default_view: 'wiki',
  term: { name: 'Default Term' },
  enrollments: [{
    enrollment_state: 'active',
    role: 'TeacherEnrollment',
    role_id: '1',
    type: 'teacher',
    user_id: '1',
  }],
})

export const courseWithSection: Template<Course> = function (defaults) {
  return course({
    ...defaults,
    sections: [section()],
  })
}

export const customColors: Template<CustomColors> = template({
  custom_colors: {
    course_1: '#fff',
  },
})
