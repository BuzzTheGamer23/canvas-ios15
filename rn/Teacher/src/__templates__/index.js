// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './folder'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './page'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './toDo'
export * from './users'
