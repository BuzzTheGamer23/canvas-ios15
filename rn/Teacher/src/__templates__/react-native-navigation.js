// @flow

import template, { type Template } from '../utils/template'

export const navigator: Template<ReactNavigator> = template({
  push: jest.fn(),
  showModal: jest.fn(),
  dismissModal: jest.fn(),
  setOnNavigatorEvent: jest.fn(),
  pop: jest.fn(),
  setTitle: jest.fn(),
  setStyle: jest.fn(),
})
