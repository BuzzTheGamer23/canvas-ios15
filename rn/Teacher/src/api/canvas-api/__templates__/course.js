/* @flow */

import template, { type Template } from '../../../utils/template'

export const course: Template<Course> = template({
  id: '1',
  name: 'Learn React Native',
  short_name: 'rn',
  course_code: 'rn 101',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  is_favorite: true,
  default_view: 'wiki',
  term: { name: 'Default Term' },
})

export const customColors: Template<CustomColors> = template({
  custom_colors: {
    course_1: '#fff',
  },
})
