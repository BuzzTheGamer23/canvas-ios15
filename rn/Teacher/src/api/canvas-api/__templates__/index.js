/* @flow */

export default {
  ...require('./course'),
}
