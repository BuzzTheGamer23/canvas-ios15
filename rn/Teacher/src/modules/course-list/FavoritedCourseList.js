// @flow

import React, { Component, PropTypes } from 'react'
import {
  Dimensions,
  View,
  Text,
  Image,
  StyleSheet,
} from 'react-native'
import Button from 'react-native-button'
import i18n from 'format-message'
import { stateToProps } from './props'
import CoursesActions from './actions'
import { connect } from 'react-redux'
import CourseList from './components/CourseList'
import NoCourses from './NoCourses'
import { route } from '../../routing'

const { width: deviceWidth } = Dimensions.get('window')

type Props = {
  navigator: ReactNavigator,
  refreshCourses: () => void,
  courses: Array<Course>,
  customColors: { [string]: string },
  error?: string,
  pending?: number,
}

export class FavoritedCourseList extends Component {
  props: Props

  static navigatorButtons = {
    rightButtons: [
      {
        title: i18n({
          default: 'Edit',
          description: 'Shown at the top of the app to allow the user to edit their course list',
        }),
        id: 'edit',
        testID: 'e2e_rules',
      },
    ],
    leftButtons: [{
      title: i18n({
        default: 'Leave Feedback',
        description: 'Shown at the top of the app to allow the user to leave feedback',
      }),
      icon: require('../../images/feedback.png'),
    }],
  }

  componentDidMount () {
    this.props.refreshCourses()
  }

  selectCourse = (course: Course) => {
    let destination = route('/courses/' + course.id)
    this.props.navigator.push(destination)
  }

  goToAllCourses = () => {
    this.props.navigator.push({
      screen: 'teacher.AllCourseList',
      title: i18n({
        default: 'All Courses',
        description: `The title of the screen showing all of a teacher's courses`,
      }),
      backButtonTitle: i18n({
        id: 'back_courses',
        default: 'Courses',
        description: 'The back button title to go from all courses to just favorited courses',
      }),
    })
  }

  renderHeader = () => {
    return (
      <View style={styles.header}>
        <View style={styles.headerTextWrapper}>
          <Image source={require('../../images/star-filled.png')} />
          <Text style={styles.headerText}>
            {i18n({
              default: 'Courses',
              description: 'The header for the favorited courses list',
            })}
          </Text>
        </View>
        <Button onPress={this.goToAllCourses} testID='course-list.see-all-btn'>
          <Text style={styles.seeAll}>
            {i18n({
              default: 'See All',
              description: 'Button to transition from favorited courses list to all courses list',
            })}
          </Text>
        </Button>
      </View>
    )
  }

  render (): React.Element<any> {
    let courses = this.props.courses.filter(course => course.is_favorite)

    if (!this.props.pending && !courses.length) {
      return (<NoCourses/>)
    }

    return (
      <CourseList
        {...this.props}
        courses={courses}
        selectCourse={this.selectCourse}
        width={deviceWidth}
        header={this.renderHeader()}
      />
    )
  }
}

const courseListShape = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  course_code: PropTypes.string.isRequired,
  short_name: PropTypes.string,
  color: PropTypes.string,
  image_download_url: PropTypes.string,
}).isRequired

FavoritedCourseList.propTypes = {
  courses: PropTypes.arrayOf(courseListShape).isRequired,
  pending: PropTypes.number,
  error: PropTypes.string,
}

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-end',
    marginTop: 8,
  },
  headerTextWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  headerText: {
    fontSize: 20,
    fontWeight: '600',
    marginLeft: 6,
  },
  seeAll: {
    fontSize: 14,
    color: '#6495ed',
    fontWeight: '500',
  },
})

let Connected = connect(stateToProps, CoursesActions)(FavoritedCourseList)
export default (Connected: Component<any, Props, any>)
