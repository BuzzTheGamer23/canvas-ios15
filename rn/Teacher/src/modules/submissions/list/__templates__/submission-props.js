// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
})
