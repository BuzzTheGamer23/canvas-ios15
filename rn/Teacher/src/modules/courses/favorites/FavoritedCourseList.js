//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  Dimensions,
  View,
  Image,
  StyleSheet,
} from 'react-native'

import { getSession } from '../../../canvas-api'
import localeSort from '../../../utils/locale-sort'
import { LinkButton } from '../../../common/buttons'
import i18n from 'format-message'
import CoursesActions from '../actions'
import { connect } from 'react-redux'
import CourseList from '../components/CourseList'
import NoCourses from '../components/NoCourses'
import type { CourseProps, CourseListDataProps } from '../course-prop-types'
import Images from '../../../images/'
import refresh from '../../../utils/refresh'
import { Heading1 } from '../../../common/text'
import Navigator from '../../../routing/Navigator'
import Screen from '../../../routing/Screen'
import branding from '../../../common/branding'
import color from '../../../common/colors'
import ActivityIndicatorView from '../../../common/components/ActivityIndicatorView'
import App from '../../app'

const { width: deviceWidth } = Dimensions.get('window')

type Props = {
  navigator: Navigator,
  refreshCourses: () => void,
  courses: Array<CourseProps>,
  error?: string,
  pending?: number,
  totalCourseCount: number,
} & RefreshProps

type FavoritedCourseListDataProps = CourseListDataProps & {
  totalCourseCount: number,
}

type State = {
  showingModal: boolean,
}

export class FavoritedCourseList extends Component {
  props: Props
  state: State

  constructor (props: Props) {
    super(props)
    this.state = { showingModal: false }
  }

  componentWillReceiveProps (newProps: Props) {
    if (!newProps.pending &&
        !newProps.error &&
        !newProps.totalCourseCount &&
        !this.state.showingModal &&
        getSession()) {
      this.props.navigator.show('/notATeacher', { modal: true })
      this.setState({
        showingModal: true,
      })
    }
  }

  showFavoritesList = () => {
    this.props.navigator.show('/course_favorites', { modal: true })
  }

  showUserCoursePreferences = (courseId: string) => {
    this.props.navigator.show(`/courses/${courseId}/user_preferences`, { modal: true })
  }

  selectCourse = (course: Course) => {
    this.props.navigator.show(`/courses/${course.id}`, { modal: true })
  }

  goToAllCourses = () => {
    this.props.navigator.show('/courses')
  }

  goToProfile = () => {
    this.props.navigator.show('/profile', { modal: true })
  }

  renderHeader = () => {
    return (
      <View style={styles.header}>
        <View style={styles.headerTextWrapper}>
          <Image source={Images.starFilled} style={{ tintColor: '#C7CDD1', height: 18, width: 18 }} testID='favorited-course-list.header-star-img'/>
          <Heading1 style={styles.headerText} testID='favorited-course-list.header-courses-lbl'>
            {i18n('Courses')}
          </Heading1>
        </View>
        <LinkButton textStyle={styles.seeAll} onPress={this.goToAllCourses} testID='favorited-course-list.see-all-btn'>
            {i18n('See All')}
        </LinkButton>
      </View>
    )
  }

  _renderComponent = () => {
    if (this.props.pending && !this.props.refreshing) {
      return <ActivityIndicatorView />
    }

    if (!this.props.pending && !this.props.courses.length && this.props.totalCourseCount) {
      return <NoCourses onAddCoursePressed={this.showFavoritesList} />
    }

    return <CourseList
      {...this.props}
      selectCourse={this.selectCourse}
      onCoursePreferencesPressed={this.showUserCoursePreferences}
      width={deviceWidth}
      header={this.renderHeader()}
      onRefresh={this.props.refresh}
    />
  }

  render () {
    let avatarURL
    const session = getSession()
    if (session) {
      avatarURL = session.user.avatar_url
    }

    const avatarConfig = {
      uri: avatarURL,
      width: 32,
      height: 32,
      borderRadius: 16,
    }

    return (
      <Screen
        navBarHidden={false}
        navBarImage={branding.headerImage}
        navBarColor={color.navBarColor}
        navBarButtonColor={color.navBarTextColor}
        navBarStyle='dark'
        rightBarButtons={[
          {
            title: i18n('Edit'),
            testID: 'favorited-course-list.edit-btn',
            action: this.showFavoritesList,
            disabled: !this.props.totalCourseCount,
          },
        ]}
        leftBarButtons={[
          {
            image: avatarURL ? avatarConfig : Images.profile,
            testID: 'favorited-course-list.profile-btn',
            action: this.goToProfile,
            accessibilityLabel: i18n('Profile'),
          },
        ]}
        >
        { this._renderComponent() }
      </Screen>
    )
  }
}

const coursePropsShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  course_code: PropTypes.string.isRequired,
  short_name: PropTypes.string,
  image_download_url: PropTypes.string,
  color: PropTypes.string,
})

FavoritedCourseList.propTypes = {
  courses: PropTypes.arrayOf(coursePropsShape).isRequired,
  pending: PropTypes.number,
  error: PropTypes.string,
}

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-end',
    paddingTop: 4,
  },
  headerTextWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  headerText: {
    marginLeft: 6,
  },
  seeAll: {
    fontWeight: '500',
  },
})

export function mapStateToProps (state: AppState): FavoritedCourseListDataProps {
  const allCourses = state.entities.courses
  const totalCourseCount = Object.keys(allCourses).length
  const { pending, error, courseRefs } = state.favoriteCourses
  const courses: Array<CourseProps> = courseRefs
    .map(ref => allCourses[ref])
    .map(({ course, color }) => ({ ...course, color }))
    .filter(App.current().filterCourse)
    .sort((c1, cs2) => localeSort(c1.name, cs2.name))

  return { pending, error, courses, totalCourseCount }
}

let Refreshed = refresh(
  props => props.refreshCourses(),
  props => props.courses.length === 0,
  props => Boolean(props.pending)
)(FavoritedCourseList)
let Connected = connect(mapStateToProps, CoursesActions)(Refreshed)
export default (Connected: Component<Props, any>)
