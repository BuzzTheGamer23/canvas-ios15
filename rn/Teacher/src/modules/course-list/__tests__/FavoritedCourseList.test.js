/* @flow */

import 'react-native'
import React from 'react'
import * as courseTemplate from '../../../api/canvas-api/__templates__/course'
import * as navigationTemplate from '../../../__templates__/react-native-navigation'
import { FavoritedCourseList } from '../FavoritedCourseList.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

jest.mock('TouchableHighlight', () => 'TouchableHighlight')

const courses = [
  courseTemplate.course({
    name: 'Biology 101',
    course_code: 'BIO 101',
    short_name: 'BIO 101',
    id: 1,
    is_favorite: true,
  }),
  courseTemplate.course({
    name: 'American Literature Psysicks foobar hello world 401',
    course_code: 'LIT 401',
    short_name: 'LIT 401',
    id: 2,
    is_favorite: false,
  }),
  courseTemplate.course({
    name: 'Foobar 102',
    course_code: 'FOO 102',
    id: 3,
    short_name: 'FOO 102',
    is_favorite: true,
  }),
]

const colors = {
  '1': '#27B9CD',
  '2': '#8F3E97',
  '3': '#8F3E99',
}

let defaultProps = {
  navigator: navigationTemplate.navigator(),
  courses,
  customColors: colors,
  refreshCourses: () => {},
  pending: 0,
}

test('renders correctly', () => {
  let tree = renderer.create(
    <FavoritedCourseList {...defaultProps} width={320} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly with wide device', () => {
  let tree = renderer.create(
    <FavoritedCourseList {...defaultProps} width={768} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('refreshCourses was called', () => {
  let didCallRefresh = false
  let refresh = () => {
    didCallRefresh = true
  }
  let tree = renderer.create(
    <FavoritedCourseList {...defaultProps} width={320} refreshCourses={ refresh } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
  expect(didCallRefresh).toBe(true)
})

test('renders no courses', () => {
  let tree = renderer.create(
    <FavoritedCourseList {...defaultProps} courses={ [] } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
