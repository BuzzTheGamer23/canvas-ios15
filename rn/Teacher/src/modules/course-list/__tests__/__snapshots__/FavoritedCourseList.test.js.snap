// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "marginHorizontal": 12,
        "overflow": "visible",
      },
    ]
  }
>
  <View>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "paddingHorizontal": 12,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={1}
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 20,
                    "fontWeight": "600",
                    "marginLeft": 6,
                  }
                }
              >
                Courses
              </Text>
            </View>
            <TouchableOpacity
              accessibilityComponentType="button"
              accessibilityTraits="button"
              activeOpacity={0.2}
              onLongPress={undefined}
              onPress={[Function]}
              onPressIn={undefined}
              onPressOut={undefined}
              style={undefined}
              testID="course-list.see-all-btn"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#368BD8",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                    Object {
                      "color": "#368BD8",
                    },
                  ]
                }
              >
                See All
              </Text>
            </TouchableOpacity>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            }
          }
        >
          <TouchableHighlight
            onLayout={[Function]}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#000",
                  "borderColor": "#e3e3e3",
                  "borderRadius": 4,
                  "borderWidth": 0.5,
                  "height": 160,
                  "overflow": "hidden",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 1,
                },
                Object {
                  "flex": 1,
                  "height": 0,
                  "margin": 12,
                },
              ]
            }
            testID="BIO 101"
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
                    }
                  }
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "#27B9CD",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0.8,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 18,
                      "position": "absolute",
                      "right": 8,
                      "top": 10,
                      "width": 18,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#27B9CD",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "marginTop": 8,
                    }
                  }
                >
                  Biology 101
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#8B969D",
                      "fontSize": 12,
                      "fontWeight": "600",
                      "marginTop": 2,
                    }
                  }
                >
                  BIO 101
                </Text>
              </View>
            </View>
          </TouchableHighlight>
          <TouchableHighlight
            onLayout={[Function]}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#000",
                  "borderColor": "#e3e3e3",
                  "borderRadius": 4,
                  "borderWidth": 0.5,
                  "height": 160,
                  "overflow": "hidden",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 1,
                },
                Object {
                  "flex": 1,
                  "height": 0,
                  "margin": 12,
                },
              ]
            }
            testID="FOO 102"
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
                    }
                  }
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "#8F3E99",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0.8,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 18,
                      "position": "absolute",
                      "right": 8,
                      "top": 10,
                      "width": 18,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#8F3E99",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "marginTop": 8,
                    }
                  }
                >
                  Foobar 102
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#8B969D",
                      "fontSize": 12,
                      "fontWeight": "600",
                      "marginTop": 2,
                    }
                  }
                >
                  FOO 102
                </Text>
              </View>
            </View>
          </TouchableHighlight>
          <View
            style={
              Object {
                "flex": 1,
                "margin": 12,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`render without courses 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Object {
          "color": "#2D3B45",
          "fontSize": 24,
          "fontWeight": "bold",
          "letterSpacing": -1,
          "lineHeight": 27,
        },
        undefined,
      ]
    }
  >
    Welcome!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Object {
          "color": "#7F91A7",
          "fontSize": 16,
          "lineHeight": 23,
        },
        Object {
          "padding": 15,
          "textAlign": "center",
        },
      ]
    }
  >
    Add a few of your favorite courses to make this place your home.
  </Text>
  <TouchableOpacity
    accessibilityComponentType="button"
    accessibilityTraits="button"
    activeOpacity={0.2}
    onLongPress={undefined}
    onPress={[Function]}
    onPressIn={undefined}
    onPressOut={undefined}
    style={
      Array [
        Object {
          "backgroundColor": "#368BD8",
          "borderRadius": 8,
          "overflow": "hidden",
          "padding": 20,
        },
        Object {
          "backgroundColor": "#368BD8",
        },
        undefined,
      ]
    }
    testID="noCourse.addCourses"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007aff",
            "fontSize": 17,
            "fontWeight": "500",
            "textAlign": "center",
          },
          null,
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "color": "#fff",
            },
            undefined,
          ],
          null,
        ]
      }
    >
      Add Courses
    </Text>
  </TouchableOpacity>
</View>
`;
